src/mixins/README.md

```diff
@@ -0,0 +1,82 @@
+# Mixins
+
+This directory contains mixins for `Application`.
+
+## Overview
+
+A mixin is a class which take the input of a base class and adds / modifies existing functions / status values and returns a new class which can be instantiated. The type of the returned class will still be the base class. Mixins let you extend the BaseClass by adding more functions to it, overriding existing ones or modifiying their behavior.
+
+## Basic Usage
+
+The general idea behind using a mixin is to pass in the BaseClass to the Mixin and create a new class. 
+
+**Example**
+```
+cosnt newClass = MixinClass(BaseClass);
+```
+
+Mixins can be nested, as such you can do the following:
+```
+cosnt newClass = MixinClass1(MixinClass2(BaseClass))
+```
+
+### LoggerMixin
+
+LoggerMixin adds capabilities to a LoopBack Next Application by adding a `.logger()` function that is used to bind a Logger class to `Context` automatically. The binding key will be `loggers.${Class.name}` where `Class.name` is the name of the Logger class being bound. Components are also able to provide their own Logger implementation which will be bound via `.component()` automatically when using this Mixin.
+
+**Example**
+```
+const LoggingApplication = LoggerMixin(Application); // we mixin the Application class from @loopback/core
+
+const app = new LoggingApplication({
+  loggers: [ColorLogger] // we can provide an array of loggers to bind automatically at startup
+});
+
+// Example Logger
+class ColorLogger {
+  log(...args: any[]) {
+    console.log('log  :', ...args);
+  }
+
+  info(...args: any[]) {
+    const data = args.join(' ');
+    console.log('\x1b[32m info : ' + data + '\x1b[0m');
+  }
+
+  warn(...args: any[]) {
+    const data = args.join(' ');
+    console.log('\x1b[33m warn : ' + data + '\x1b[0m');
+  }
+
+  error(...args: any[]) {
+    const data = args.join(' ');
+    console.log('\x1b[31m error: ' + data + '\x1b[0m');
+  }
```

Comment:
In order to keep these examples shorter, I am proposing to reduce the number of logger methods to two: `log` and `error`. (This will require changes in the source code too.)

Reply:
Code change will only apply to test cases as the mixin doesn't supply a Logger implementation.